sail/src/main/java/org/apache/rya/rdftriplestore/inference/InferenceEngine.java

```diff
@@ -220,163 +229,163 @@ public void refreshGraph() throws InferenceEngineException {
-            				  break;
-            			  }
-            			  iter2.close();
-            		  }
-            		  iter2 = RyaDAOHelper.query(ryaDAO, vf.createURI(st.getObject().stringValue()), RDF.REST,
-                  			null, conf);
-            		  if (iter2 != null){
-            			  while (iter2.hasNext()){
-            				  Statement iter2Statement = iter2.next();
-            				  anonPropNode = iter2Statement.getObject();
-            				  break;
-            			  }
-            			  iter2.close();
-            			  if (anonPropNode != null){
-            				  iter2 = RyaDAOHelper.query(ryaDAO, vf.createURI(anonPropNode.stringValue()), RDF.FIRST,
-                            			null, conf);
-            				  while (iter2.hasNext()){
-                				  Statement iter2Statement = iter2.next();
-                				  propURI = iter2Statement.getObject();
-                				  break;
-                			  }
-                			  iter2.close();
-            			  }
-            		  }
-            		  if (!integerValue.isEmpty() && propURI!=null) {
-            			  try {
-                			  int indexValue = Integer.parseInt(integerValue);
-                			  URI chainPropURI = vf.createURI(propURI.stringValue());
-                			  orderedProperties.put(indexValue, chainPropURI);
-            			  }
-            			  catch (Exception ex){
-            				  // TODO log an error here
-            				  
-            			  }
-            		  }
-            		}
-            	} finally{
-            		if (iter != null){
-            			iter.close();
-            		}
-            	}
-            	List<URI> properties = new ArrayList<URI>();
-            	for (Map.Entry<Integer, URI> entry : orderedProperties.entrySet()){
-            		properties.add(entry.getValue());
-            	}
-            	propertyChainPropertyToChain.put(propertyChainProperty, properties);
+            for (final URI propertyChainProperty : propertyChainPropertiesToBNodes.keySet()){
+                final URI bNode = propertyChainPropertiesToBNodes.get(propertyChainProperty);
+                // query for the list of indexed properties
+                iter = RyaDAOHelper.query(ryaDAO, bNode, vf.createURI("http://www.w3.org/2000/10/swap/list#index"),
+                        null, conf);
+                final TreeMap<Integer, URI> orderedProperties = new TreeMap<Integer, URI>();
+                // TODO refactor this.  Wish I could execute sparql
+                try {
+                    while (iter.hasNext()){
+                      final Statement st = iter.next();
+                      final String indexedElement = st.getObject().stringValue();
+                      log.info(indexedElement);
+                      CloseableIteration<Statement, QueryEvaluationException>  iter2 = RyaDAOHelper.query(ryaDAO, vf.createURI(st.getObject().stringValue()), RDF.FIRST,
+                                null, conf);
```

Comment:
Yea, sorry.  It does make it hard.  Mostly refreshIntersectionOf() was the biggest change in here.